legend-engine-authentication-protocol/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/auth/ConnectionProvider.java

```diff
@@ -0,0 +1,10 @@
+package org.finos.legend.engine.protocol.pure.v1.model.packageableElement.auth;
+
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.authentication.AuthenticationSpec;
+import org.finos.legend.engine.shared.core.identity.Identity;
+
+public abstract class ConnectionProvider<T> {
+
+    public abstract T makeConnection(Object connectionSpec, AuthenticationSpec authenticationSpec, Identity identity) throws Exception;
```

Comment:
why is connectionSpec an Object, seems very weak ref

Reply:
Fixed. Added abstract class ConnectionSpec